tap_google_ads/streams.py

```diff
@@ -56,6 +58,29 @@ def get_request_timeout(config):
         request_timeout = DEFAULT_REQUEST_TIMEOUT
     return request_timeout
 
+
+def get_period_len(config):
+    """Constructs the period length value from config and error on invalid values"""
+
+    period_len_days = config.get("period_len_days") or "1"
+    period_len_weeks = config.get("period_len_weeks") or "0"
+    period_len_months = config.get("period_len_months") or "0"
+    period_len_years = config.get("period_len_years") or "0"
+
+    try:
+        period_len = relativedelta(
+            days = int(period_len_days),
+            weeks = int(period_len_weeks),
+            months = int(period_len_months),
+            years = int(period_len_years)
```

Comment:
`#idea` support 1 parameter instead of 4:
```
config.json
{
  ...
  "split_by_periods": {
    "unit": "days|weeks|months|years",
    "amount": int
  }
}
``` 
`#q` do we really need to have a custom `amount`?  In what cases will we use it? I guess we can always use 1, can't we?

Reply:
Good question. Looks like some #dk is missing. Custom amount is needed for backward compatibility with the current version so the caller can instruct the tap to fetch all data in one request by `period_len_days = end_date - start_date + 1`.

Reply:
One more usage of custom "amount" - Google Ads provides "Quarter" as a date dimension. So we should use "period_len_months=3" to support that.

Reply:
1. Renamed `period_len` to `split_by_period`
2. Made the parameter a dict of `days, weeks, months, years`.
3. Made code request the whole period in one request when `split_by_period` value is not present in the config to preserve backward compatibility.

Reply:
ok